server/pulp/server/managers/status.py

```diff
@@ -51,9 +54,14 @@ def get_broker_conn_status():
     # explicit connection and then release it.
     # See https://github.com/celery/kombu/issues/432 for more detail.
     try:
-        conn = celery.connection()
         conn.connect()
+        transport = conn._transport
         conn.release()
+
+        # An explicit call to __del__ to release file descriptors
+        # https://pulp.plan.io/issues/2124#note-9
+        transport.__del__()
```

Comment:
Yeah, the tl;dr of python for me on this has always been "just don't bother implementing `__del__`, and talk anyone you see using it out of it."

I think this is a tricky problem, and whatever you come up with to improve on this is probably fine. LGTM, but if you want to take a few minutes and chat about it I'll be available. :)